data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/InactiveAcknowledgementSetManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.acknowledgements;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import java.util.function.Consumer;
+import java.time.Duration;
+
+public class InactiveAcknowledgementSetManager implements AcknowledgementSetManager {
+    private static InactiveAcknowledgementSetManager theInstance;
+
+    public static InactiveAcknowledgementSetManager getInstance() {
+        if (theInstance == null) {
+            theInstance = new InactiveAcknowledgementSetManager();
+        }
+        return theInstance;
+    }
+    
+    public AcknowledgementSet create(final Consumer<Boolean> callback, final Duration timeout) {
+        throw new UnsupportedOperationException("create operation not supported");
+    }
+
+    public void acquireEventReference(final Event event) {
```

Comment:
We spoke about this offline. Since this code adds to a `Set`, we want to avoid performing adding these to the `Set` if there is no need to track the references.

The `RouterCopyRecordStrategy` is using a condition to check if it is the `InactiveAcknowledgementSetManager` and if so, it will skip the entire logic of adding to the `Set`. It will also avoid calling `acquireEventReference` altogether.

So the current approach will help avoid a possible performance penalty when end-to-end acknowledgements are not is use.

We can circle back on this later to see if there is a cleaner way to handle this. But, for now, this code will work.